### DISCUSSION TITLE
Question: Linked Integration Runtimes

### DISCUSSION THREAD

**Discussion by mikedferris** [Azure-Player/azure.datafactory.tools#160] [Q&A] [answered]
If the target data factory shares integration runtimes with the source, whereby the IR is Shared on the source DF and Linked on the target, what's the best way to handle this in the deployment?

I wondered about using a config.csv file for this, but the value replacement isn't simple, so I wasn't sure if it would work.

For example, the source IR will have the following properties:

    "properties": {
        "type": "SelfHosted",
        "description": "IR description"
    }

But the target Linked IR will look like this:

"properties": {
	"type": "SelfHosted",
	"typeProperties": {
		"linkedInfo": {
			"resourceId": "/subscriptions/xxxxxxxxxxxxxxxx/resourcegroups/xxxxxx/providers/Microsoft.DataFactory/factories/xxxxx/integrationruntimes/xxxxx-SelfHostedIR",
			"authorizationType": "Rbac"
		}
	}
}

Can the CSV file handle such a complex replacement, and if so, how should the " symbols be incorporated?

Thank you

**Comment by mikedferris** [comment, marked as answer]
Further investigation shows this to be a non-issue. In the target Data Factory if the Linked IRs are added with the exact same name and then deployment ignores Integration Runtimes via the Filtering text parameter then Linked Services pick up the correct IR in the target Data Factory.

**Comment by EkueAF** [comment]
Hello evryone,
I found this in microsoft documentation : https://learn.microsoft.com/en-us/azure/data-factory/continuous-integration-delivery

_"Integration runtimes and sharing. Integration runtimes don't change often and are similar across all stages in your CI/CD. So Data Factory expects you to have the same name, type and sub-type of integration runtime across all stages of CI/CD. If you want to share integration runtimes across all stages, consider using a ternary factory just to contain the shared integration runtimes. You can use this shared factory in all of your environments as a linked integration runtime type.

 Note

The integration runtime sharing is only available for self-hosted integration runtimes. Azure-SSIS integration runtimes don't support sharing._"

My question is : Should we have to create a ternary factory using adftools module for Sahred Integration Runtime deployment?
